docs/example_navigation_ros_codegen.md

```diff
@@ -0,0 +1,540 @@
+    if _u_ref is None:
+        _u_ref = cs.DM.zeros(_u.shape)
+    dx = _x - _x_ref
+    du = _u - _u_ref
+    return cs.mtimes([dx.T, Q, dx]) + cs.mtimes([du.T, R, du])
+
+
+# The terminal cost for x
+def terminal_cost(_x, _x_ref=None):
+    if _x_ref is None:
+        _x_ref = cs.DM.zeros(_x.shape)
+    dx = _x - _x_ref
+    return cs.mtimes([dx.T, QN, dx])
+
+
+x_0 = cs.MX.sym("x_0", NX)
+x_ref = cs.MX.sym("x_ref", NX)
+u_k = [cs.MX.sym('u_' + str(i), NU) for i in range(N)]
+
+# Create the cost function
+x_t = x_0
+total_cost = 0
+
+for t in range(0, N):
+    total_cost += stage_cost(x_t, u_k[t], x_ref)  # update cost
+    x_t = dynamics_dt(x_t, u_k[t])  # update state
+
+total_cost += terminal_cost(x_t, x_ref)  # terminal cost
+
+optimization_variables = []
+optimization_parameters = []
+
+optimization_variables += u_k
+optimization_parameters += [x_0]
+optimization_parameters += [x_ref]
+
+optimization_variables = cs.vertcat(*optimization_variables)
+optimization_parameters = cs.vertcat(*optimization_parameters)
+
+umin = [-2.0, -1.0] * N  # - cs.DM.ones(NU * N) * cs.inf
+umax = [2.0, 1.0] * N  # cs.DM.ones(NU * N) * cs.inf
+
+bounds = og.constraints.Rectangle(umin, umax)
+
+problem = og.builder.Problem(optimization_variables,
+                             optimization_parameters,
+                             total_cost) \
+    .with_constraints(bounds)
+
+ros_config = og.config.RosConfiguration() \
+    .with_package_name("open_nmpc_controller") \
+    .with_node_name("open_mpc_controller_node") \
+    .with_rate((int)(1.0/sampling_time)) \
+    .with_description("Cool ROS node.")
+
+build_config = og.config.BuildConfiguration()\
+    .with_build_directory("optimization_engine")\
+    .with_build_mode("release")\
+    .with_build_c_bindings() \
+    .with_ros(ros_config)
+
+meta = og.config.OptimizerMeta()\
+    .with_optimizer_name("mpc_controller")
+
+solver_config = og.config.SolverConfiguration()\
+    .with_tolerance(1e-5)
+
+builder = og.builder.OpEnOptimizerBuilder(problem,
+                                          meta,
+                                          build_config,
+                                          solver_config)
+builder.build()
+```
+
+The above program will generate a parametric optimizer at `optimization_engine/mpc_controller`. The ROS package will be in `optimization_engine/mpc_controller/open_nmpc_controller`.
+
+## Create a ROS workspace for our new package
+
+We must create a ROS catkin workspace and place our package inside the `src` folder. We will do this by creating a symbolic link. After that, we can build and test our package.
+
+```bash
+source /opt/ros/melodic/setup.bash
+cd ~/open_ros_codegen/
+mkdir -p ~/open_ros_codegen/catkin_ws/src
+cd ~/open_ros_codegen/catkin_ws/src
+ln -s ~/open_ros_codegen/nmpc_open/optimization_engine/mpc_controller/open_nmpc_controller .
+cd ~/open_ros_codegen/catkin_ws
+catkin build
+source ~/open_ros_codegen/catkin_ws/devel/setup.bash
```

Comment:
Is this `source` necessary?

Reply:
Yes, in order to be sure that the newly generated packages are loaded.